source/develop/release-management/features/storage/hotplugdisk.html.md

```diff
@@ -17,65 +17,42 @@ feature_status: Released
-*   Status: Almost finished
-*   Last updated date: ...
+Supported since oVirt 3.1. VirtIO-SCSI support added in oVirt 3.3.
 
 ### Detailed Description
 
-The following feature will allow to hot plug/unplug of virtIO disks on running vm, also in scope of that feature will be added a new attribute to disk which will allow to run vm even the following disk is unaccesiable.
-This feature will be restricted to operating systems that support this action – eg. RHEL 5/6, Windows Server 2003 and Windows server 2008.
-
-#### Entity Description
-
-The following changes on disk entity will be done: A new field should be added to the table image_vm_map which will indicate status of disk (plugged, unplugged) for current vm
-A new field should be added to the table images, type boolean, which will indicate if a disk is plugged or unplugged
-
-#### CRUD
-
-No new operation will be added, an update will be done by using of existing stored procedures
+The following feature will allow to hot plug/unplug disks to a running VM. Note that this feature will require the guest's cooperation, so it will be restricted to operating systems that support it, e.g.m RHEL 5 and above, Windows Server 2003 and above, etc.
 
 #### User Experience
 
 The following UI mockup contain guidelines for VMs->Disks sub-tab:
 
 ![](Vm_disks_hotplug.png "Vm_disks_hotplug.png")
 
-#### Installation/Upgrade
-
-An upgrade script will add a missed fields to the table disks and view images. The tables can be changed in the future, because of changes at schema which are currently is done due to development of different features.
-
 #### User work-flows
 
-New actions will allow the user to plug or unplug a disk to/from a running VM, it will be allowed only for disks which were added to vm by AddDiskToVMCommand . The feature will be allowed only on 3.1 clusters and above
-Via UI or the API the user will have an option to plug or unplug a disk of a running VM.
-The action will be allowed in the following flows:
+New actions will allow the user to plug or unplug a disk to/from a running VM in 3.1 clusters and above.
+
 HotPlug will be allowed only on:
 
-1. Operation system of guest is supports operation
-2. VirtIO disk
+1. The guest OS supports the operation
+2. VirtIO/VirtIO-SCSI disk
 3. Disk should be unplugged
-4. VM should be in status Up
+4. VM should be up
 
 HotUnPlug will be allowed in the following cases:
 
-1. Operation system of guest is supports operation
-2. VirtIO disk
-3. VM should be in status Up
-4. Disk should be plugged
+1. The guest OS supports the operation
+2. VirtIO/VirtIO-SCSI disk
+3. Disk should be plugged
+4. VM should be up
 
-In order to perform an operation new verbs will be added at VDSM side:
-hotplug and hotunplug with the following dictionary to pass:
-A call to vdsm will be done in synchronious way
+In order to perform this operation new verbs will be added to the VDSM side:
+`hotplug` and `hotunplug` with the following argument dictionary:
```

Comment:
VM.hotplugDisk , VM.hotunplugDisk.
The parameters are not exactly a dictionary, it is VDSM types as described in the schema-
hotplugDisk gets:
{'vmID': 'UUID', 'params': 'HotplugDiskParams'}

hotunplugDisk gets:
{'vmID': 'UUID', 'params': 'HotplugDiskParams'}

These types have other types inside, and it is all detailed in VDSM's schema. anyway it is not what's written here. Question is if we want to write it all down here or refer to elsewhere.